api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -348,6 +348,10 @@ message HttpConnectionManager {
     repeated HttpFilter filters = 2;
   };
   repeated UpgradeConfig upgrade_configs = 23;
+
+  // If true, the order of encoder filters will be reversed to that of filters
+  // configured in the HTTP filter chain.
+  bool reverse_encoders_order = 27;
```

Comment:
Should we make this BoolValue and change the default to true in near future (after next release)?

Reply:
Done with the renaming.

@htuch would you recommend to add the "deprecated" annotation in this change or in another PR when we deprecate the field in that release?

 @lizan what would be the difference between bool and BoolValue in this case?

Reply:
bool is always default to false and you can't differentiate false and unset. with BoolValue you know if it is unset or true or false.

Reply:
The question is what will be the process when we remove this deprecated field. If at that moment this will be true (true is expected behavior, right?) then BoolValue makes more sense. @htuch?

Reply:
hmm, the h1_capture_direct_response_fuzz_test test is hitting tsan failure after I set the bool filed [depreated = true]. I am wondering if the BoolValue would help with the failure.

https://circleci.com/gh/envoyproxy/envoy/108918?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
It is a flaky one and fix in recent commit, merge master would help.

Reply:
Merging does help, but looks like I rebased it incorrectly and the version history is corrupted :(

Any suggestions to fix this? I guess I can close this PR and create a new one, but wonder if there is a simple solution.

Reply:
Yes you shouldn't do `git rebase` but a simple `git merge upstream/master`. you can checkout the original commit and do git merge again, and then force push to fix this PR, or close and create a new PR.

Reply:
It works! thanks for the suggestion!